Sources/Testing/ExitTests/StatusAtExit.swift

```diff
@@ -0,0 +1,73 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024–2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+private import _TestingInternals
+
+/// An enumeration describing possible status a process will yield on exit.
+///
+/// You can cast an instance of this type to an instance of
+/// ``ExitTest/Condition`` using ``ExitTest/Condition/init(_:)``. That value
+/// can then be used to describe the condition under which an exit test is
+/// expected to pass or fail by passing it to
+/// ``expect(exitsWith:observing:_:sourceLocation:performing:)`` or
+/// ``require(exitsWith:observing:_:sourceLocation:performing:)``.
+@_spi(Experimental)
+#if SWT_NO_PROCESS_SPAWNING
+@available(*, unavailable, message: "Exit tests are not available on this platform.")
+#endif
+public enum StatusAtExit: Sendable {
```

Comment:
(Separate from the name of the type) Maybe we should make this a `struct` with a `Kind` enum. I always worry about exposing enums as public API since they're less flexible if we need to add another stored value which is common to all cases. Just to help future-proof

Reply:
As I discussed with @natecook1000: this structure directly models the values reported by processes when they terminate. There is approximately zero chance that the platforms we support today are going to come up with a whole new kind of process exit status, and if they do, they'll need to report it via an entirely new mechanism as `siginfo_t`, `wait()`, etc. have no extensibility.